### DISCUSSION TITLE
Way to bypass setValue functions when calling getValues()

### DISCUSSION THREAD

**Discussion by kwinyyyc** [react-hook-form/react-hook-form#6218] [Q&A]
I have a use case to reset the form isDirty status no matter the form submission is success or not.

In the form I have used `setValueAs` to encode a textfield and `setValueAs` to convert a option selection from `{ "key": 1, "text": "Test" }` to just the key which is `1`

So when I am trying to call `reset(getValues())` at the beginning of form submit, the value of the textfield would become a encoded value and the option field become not mappable back to the selection.

Is it possible to call the getValues without triggering the `setValue` methods and get back the original raw values?

Thanks.

**Comment by bluebill1049** [comment]
skip `setValueAs` and build your own controlled component?

**Reply by kwinyyyc** [reply]
Any hints on how to do that with custom component?
If I wrapped the textfield within `<Controller>`, I still could not emit the encoded value only when the form is submitted and reset the isDirty state?

**Reply by bluebill1049** [reply]
hmmm i see what do you mean now, I think you may have to do your transform at the submit level.
